src/bpftrace.cpp

```diff
@@ -724,8 +726,86 @@ void perf_event_lost(void *cb_cookie, uint64_t lost)
   bpftrace->out_->lost_events(lost);
 }
 
-std::unique_ptr<AttachedProbe> BPFtrace::attach_probe(Probe &probe, const BpfOrc &bpforc)
+std::vector<std::unique_ptr<AttachedProbe>> BPFtrace::attach_usdt_probe(
+    Probe &probe,
+    std::tuple<uint8_t *, uintptr_t> func,
+    int pid,
+    bool file_activation)
+{
+  std::vector<std::unique_ptr<AttachedProbe>> ret;
+
+  // File activation works by scanning through /proc/*/maps and seeing
+  // which processes have the target executable in their address space
+  // with execute permission. If found, we will try to attach to each
+  // process we find.
+  if (file_activation && probe.path.size())
+  {
+    glob_t globbuf;
+    if (::glob("/proc/[0-9]*/maps", GLOB_NOSORT, nullptr, &globbuf))
+      throw std::runtime_error("failed to glob");
+
+    const char *p;
+    if (!(p = realpath(probe.path.c_str(), nullptr)))
+    {
+      std::cerr << "Failed to resolve " << probe.path << std::endl;
+      return ret;
+    }
+    std::string resolved(p);
+
+    for (size_t i = 0; i < globbuf.gl_pathc; ++i)
+    {
+      std::string path(globbuf.gl_pathv[i]);
+      std::ifstream file(path);
+      if (file.fail())
+        throw std::runtime_error("failed to open " + path);
```

Comment:
Won't this cause an issue if a pid exits between the glob and now?

Reply:
Hmm good point, guess we have to fail silently here